src/components/CartButton/CartButton.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import styles from './CartButton.module.css';
+import { useGlobalContext } from '../../context/Provider';
+
+function CartButton({ id, numElements, stock, url, name, price, alt }) {
+    const { addItemToCart, itemsInCart } = useGlobalContext();
+
+    const isDisabled = () => {
+        const disabled =
+            stock == 0
+                ? true
+                : itemsInCart &&
+                  id in itemsInCart &&
+                  itemsInCart[id].quantity >= stock
+                ? true
+                : false;
```

Comment:
we can use the conditions and remove the ternary to get the value of `disabled`:

```suggestion
        const disabled = stock === 0 || (
          itemsInCart &&
          id in itemsInCart &&
          itemsInCart[id].quantity >= stock
        );
```